lib/sparrow/pools_warden.ex

```diff
@@ -94,33 +94,43 @@ defmodule Sparrow.PoolsWarden do
       end)
   end
 
-  @spec handle_info(any, :ok) :: {:noreply, :ok}
-  def handle_info(unknown, _state) do
+  @impl GenServer
+  def handle_info({:DOWN, _ref, :process, pid, reason}, state) do
+    {pid, pool_type, pool_name, tags} = List.keyfind(state, pid, 0)
```

Comment:
do you duplicate `ets` stored data in process state? And why?

Reply:
We don't store PID info in ets table - I'm not sure if we should. Moreover, for me it made sense to store this kind of information in state and we use maps for it to be quick